src/appleseed/renderer/kernel/intersection/intersector.cpp

```diff
@@ -1,389 +1,389 @@
-    const ShadingRay::RayInfoType ray_info(ray);
-
-    // Refine and offset the previous intersection point.
-    if (parent_shading_point &&
-        parent_shading_point->hit_surface() &&
-        !(parent_shading_point->m_members & ShadingPoint::HasRefinedPoints))
-        parent_shading_point->refine_and_offset();
-
-    // Retrieve assembly tree.
-    const AssemblyTree& assembly_tree = m_trace_context.get_assembly_tree();
-
-    // Check the intersection between the ray and the assembly tree.
-    AssemblyTreeProbeIntersector intersector;
-    AssemblyLeafProbeVisitor visitor(
-        assembly_tree,
-        m_triangle_tree_cache,
-        m_curve_tree_cache,
-#ifdef APPLESEED_WITH_EMBREE
-        m_embree_scene_cache,
-#endif
-        parent_shading_point
-#ifdef FOUNDATION_BVH_ENABLE_TRAVERSAL_STATS
-        , m_triangle_tree_traversal_stats
-#endif
-        );
-    intersector.intersect_no_motion(
-        assembly_tree,
-        ray,
-        ray_info,
-        visitor
-#ifdef FOUNDATION_BVH_ENABLE_TRAVERSAL_STATS
-        , m_assembly_tree_traversal_stats
-#endif
-        );
-
-    return visitor.hit();
-}
-
-void Intersector::make_triangle_shading_point(
-    ShadingPoint&                       shading_point,
-    const ShadingRay&                   shading_ray,
-    const Vector2f&                     bary,
-    const AssemblyInstance*             assembly_instance,
-    const Transformd&                   assembly_instance_transform,
-    const size_t                        object_instance_index,
-    const size_t                        primitive_index,
-    const TriangleSupportPlaneType&     triangle_support_plane) const
-{
-    // This helps finding bugs if make_surface_shading_point()
-    // is called on a previously used shading point.
-    debug_poison(shading_point);
-
-    // Context.
-    shading_point.m_texture_cache = &m_texture_cache;
-    shading_point.m_scene = &m_trace_context.get_scene();
-    shading_point.m_ray = shading_ray;
-
-    // Primary intersection results.
-    shading_point.m_primitive_type = ShadingPoint::PrimitiveTriangle;
-    shading_point.m_bary = bary;
-    shading_point.m_assembly_instance = assembly_instance;
-    shading_point.m_assembly_instance_transform = assembly_instance_transform;
-    shading_point.m_assembly_instance_transform_seq = &assembly_instance->transform_sequence();
-    shading_point.m_object_instance_index = object_instance_index;
-    shading_point.m_primitive_index = primitive_index;
-    shading_point.m_triangle_support_plane = triangle_support_plane;
-
-    // Available on-demand results: none.
-    shading_point.m_members = 0;
-}
-
+
+//
+// This source file is part of appleseed.
+// Visit https://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2010-2013 Francois Beaune, Jupiter Jazz Limited
+// Copyright (c) 2014-2018 Francois Beaune, The appleseedhq Organization

```

Comment:
2019

Reply:
At some point I'll do a pass on all files to update copyright notices. Usually I do that at the beginning of each year but this time I've been lazy. Might as well wait until January 2020 at this point :)